app/recurring_job.go

```diff
@@ -171,7 +171,7 @@ func recurringJob(c *cli.Context) error {
 			})
 			log.Info("Creating job")
 
-			snapshotName := types.GetCronJobNameForRecurringJob(jobName) + "-" + util.RandomID()
+			snapshotName := sliceStringSafely(types.GetCronJobNameForRecurringJob(jobName), 0, 8) + "-" + util.UUID()
```

Comment:
> Why we need this specific modification? or just have a createSnapshotName util function to use here.

We want to have a prefix to easily identify that the snapshot belongs to the recurring job. We also want the snapshot name to have enough entropy

Reply:
> Will this affect the existing recurring jobs?

Yes, it will but I don't see a problem. Are you foreseeing a problem?

Reply:
Not yet. The snapshot cleanup in this job relies on the label rather than this naming format. But it's better to have a test for this part:
1. Launch the old Longhorn system
2. Create a recurring job then wait for the recurring snapshot creation.
3. Upgrade to the test version
4. Update the recurring job so that the cron job will be recreated with the new image
5. Wait and check if the old snapshots can be cleaned up as usual.

Reply:
Tested. The old snapshots were cleaned up correctly